dataflux_pytorch/dataflux_mapstyle_dataset.py

```diff
@@ -232,3 +234,18 @@ def _get_or_create_storage_client(self):
         if self.storage_client is not None:
             return self.storage_client
         return storage.Client(project=self.project_name)
+
+    def __getstate__(self):
+        # Copy the object's state from self.__dict__ which contains
+        # all our instance attributes. Use the dict.copy()
+        # method to avoid modifying the original state.
+        state = self.__dict__.copy()
+        # Remove the unpicklable entries.
+        del state['storage_client']
+        return state
+
+    def __setstate__(self, state):
+        # Restore instance attributes.
+        self.__dict__.update(state)
+        # Create the storage client.
+        self.storage_client = storage.Client(project=self.project_name)
```

Comment:
This looks like a great change! Did you run any real workload tests to see what sort of time improvements we see?